libs/@guardian/source-react-components-development-kitchen/src/age-warning/AgeWarning.tsx

```diff
@@ -0,0 +1,97 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import type { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+import {
+	from,
+	palette,
+	textSans,
+	visuallyHidden,
+} from '@guardian/source-foundations';
+import { darkModeCss } from './styles';
+
+export interface AgeWarningProps {
+	/**
+	 * the age to be shown in the warning. e.g. 5 years
+	 */
+	age: string;
+	/**
```

Comment:
Looks amazing Marjan. Do we want to add some sort of logic so that if passed a unix time stamp the component could format the age appropriately?

Reply:
Thanks Tom for looking, the age prop is a string that would be shown in the warning as it is. For example if the age is `5 years` and the warningPrefix is not given, the message that's shown is `This article is more than 5 years old`

Reply:
Gotcha. Sorry - what I meant was that instead of passing in a string for the age, you could pass in a date and the component would figure out what the string should be. Essentially, we move the logic you have wrote in your AR pull request to this one: https://github.com/guardian/dotcom-rendering/pull/6686/files#diff-6c405a7ab45efe04c2c48734531cdcd4a51dae5ba56f381ab75afddd0bc79297R14

That could then mean that both DCR and AR will use exactly the same logic for figuring out dates, which could reduce code and enable consistency? Just a thought.

Reply:
Great suggestion @tkgnm 
For visibility, we had a group chat about this, and decided to create a health ticket for this improvement. So hopefully soon we'll move the age calculation logic to the AgeWarning component :)